testsuite/openmodelica/fmi/ModelExchange/2.0/fmi_attributes_09.mos

```diff
@@ -0,0 +1,144 @@
+  function foo
+    input Real a;
+    input Real b;
+    output Real c;
+  algorithm
+    if b > 2.0 then
+      c := foo(a, b-1.0);
+    else
+      c := a;
+    end if;
+  end foo;
+
+  input Real u;
+  Real x;
+  output Real y;
+  Real a, b;
+  parameter Real z=10;
+  Real z1 = foo(u, x) + 1.0;
+  Real z2 = foo(u, x) + 2.0;
+initial equation
+  x = 0.0;
+equation
+  a = 2.0*u;
+  b = 0.5*a;
+  der(x) = b + z;
+  y = x + z;
+end fmi_attributes_09;
+"); getErrorString();
+
+translateModelFMU(fmi_attributes_09); getErrorString();
+
+// unzip to console, quiet, extra quiet
+system("unzip -cqq fmi_attributes_09.fmu modelDescription.xml > fmi_attributes_09_tmp.xml"); getErrorString();
+system("sed -n \"/<ModelVariables>/,/<\\/ModelVariables>/p\" fmi_attributes_09_tmp.xml > fmi_attributes_09.xml"); getErrorString();
+readFile("fmi_attributes_09.xml"); getErrorString();
+
+system("sed -n \"/<ModelStructure>/,/<\\/ModelStructure>/p\" fmi_attributes_09_tmp.xml > fmi_attributes_09.xml"); getErrorString();
+readFile("fmi_attributes_09.xml"); getErrorString();
+
+// Result:
+// true
+// ""
+// "fmi_attributes_09.fmu"
+// ""
+// 0
+// ""
+// 0
+// ""
+// "  <ModelVariables>
+//   <!-- Index of variable = \"1\" -->
+//   <ScalarVariable
+//     name=\"x\"
+//     valueReference=\"0\"
+//     >
+//     <Real/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"2\" -->
+//   <ScalarVariable
+//     name=\"der(x)\"
+//     valueReference=\"1\"
+//     >
+//     <Real derivative=\"1\"/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"4\" -->
+//   <ScalarVariable
+//     name=\"a\"
+//     valueReference=\"3\"
+//     >
+//     <Real/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"5\" -->
+//   <ScalarVariable
+//     name=\"b\"
+//     valueReference=\"4\"
+//     >
+//     <Real/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"6\" -->
+//   <ScalarVariable
+//     name=\"u\"
+//     valueReference=\"5\"
+//     causality=\"input\"
+//     >
+//     <Real start=\"0.0\"/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"7\" -->
+//   <ScalarVariable
+//     name=\"y\"
+//     valueReference=\"6\"
+//     causality=\"output\"
+//     >
+//     <Real/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"8\" -->
+//   <ScalarVariable
+//     name=\"z1\"
+//     valueReference=\"7\"
+//     >
+//     <Real/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"9\" -->
+//   <ScalarVariable
+//     name=\"z2\"
+//     valueReference=\"8\"
+//     >
+//     <Real/>
+//   </ScalarVariable>
+//   <!-- Index of variable = \"10\" -->
+//   <ScalarVariable
+//     name=\"z\"
+//     valueReference=\"9\"
+//     variability=\"fixed\"
+//     causality=\"parameter\"
+//     >
+//     <Real start=\"10.0\"/>
+//   </ScalarVariable>
+//   </ModelVariables>
+// "
+// ""
+// 0
+// ""
+// "  <ModelStructure>
+//     <Outputs>
+//       <Unknown index=\"7\" dependencies=\"1\" dependenciesKind=\"dependent\" />
```

Comment:
Due to hidden variables, index 7 is actually index 6.

Reply:
this should be fixed 

> TODO:
> 
> * [ ]  Fix indices
> * [ ]  Fix model structure section

This is fixed now